### PR TITLE
Revise lifecycle scripts to avoid word splitting

### LINKED ISSUE #526: UI Install Scripts fails with password that has a space
**VIC-UI version:** 1.4

**Platform details:** Installing from VCSA

**Additional details:**

When I was running the VCSA UI plugin install script I was getting the error message that it would not be install and "--company must be specified". I tried to run the uninstall script and it gave a error that I was passing invalid parameters which it listed as being the last half of my password after a space.

I then created a new administrator user that had no spaces in it and tried again to install and it worked just fine.

### PR DESCRIPTION
**Pull request by zjs**
Update the logic in `install.sh`, `upgrade.sh`, and `uninstall.sh` to use double quotes with variables and follow other Bash best practices to avoid word splitting, globbing, and some unnecessary use of subshells.

Additional cleanup to these scripts to reduce duplication, further reduce use of subshells, and avoid masking return values is possible, but is not included in this change to limit scope and risk.

---

Fixes #526

PR acceptance checklist:

- [ ] All unit tests pass
- [ ] All e2e tests pass
- [ ] Unit test(s) included*
- [ ] e2e test(s) included*
- [ ] Screenshot attached and UX approved*

 *if applicable, add n/a if not

### CODE REVIEW

File: scripts/VCSA/upgrade.sh [hunk 1]
```diff
@@ -101,32 +100,29 @@ if [[ ${VIC_UI_HOST_URL: -1: 1} != "/" ]] ; then
...
-        read -p "Enter SHA-1 thumbprint of target VC: " VC_THUMBPRINT
-    fi
+    while true; do
+        IFS="" read -r -p "Are you sure you trust the authenticity of this host (yes/no)? " SHOULD_ACCEPT_VC_FINGERPRINT
+        case "${SHOULD_ACCEPT_VC_FINGERPRINT}" in
+            [Yy][Ee][Ss] ) return;;
+            [Nn][Oo] ) IFS="" read -r -p "Enter SHA-1 thumbprint of target VC: " VC_THUMBPRINT; return;;
+            * ) echo 'Please answer either "yes" or "no".';;
+        esac
+    done
```
Comment (lcastellano) [inline scripts/VCSA/upgrade.sh]: It looks like some code is duplicated. Is it possible to have a common utils.sh?

Comment (zjs) [reply scripts/VCSA/upgrade.sh]: As discussed in person, I'd like to do this eventually, but don't want to make large changes in this area until the UI plugin automation project is complete.

File: scripts/VCSA/install.sh [hunk 2]
```diff
@@ -74,62 +74,56 @@ fi
 
 # load configs variable into env
 while IFS='' read -r line; do
```
Comment (hickeng) [inline scripts/VCSA/install.sh]: why on earth isn't this just sourcing the file?

File: scripts/VCSA/uninstall.sh [hunk 3]
```diff
@@ -27,7 +27,7 @@ read_vc_information () {
                 VCENTER_ADMIN_PASSWORD=$OPTARG
                 ;;
             *)
-                echo Usage: $0 [-i vc_ip] [-u vc admin_username] [-p vc_admin_password] >&2
+                echo "Usage: $0 [-i vc_ip] [-u vc admin_username] [-p vc_admin_password]" >&2
```
Comment (hickeng) [inline scripts/VCSA/uninstall.sh]: `common.sh` file so this isn't duplicated between install/uninstall. Or too much effort vs fixing the plugin install properly?

**Pull request merged**